R/validation_rules.R

```diff
@@ -43,59 +43,78 @@ validate_ees_id <- function(element_id, level = "publication") {
       )
     )
   }
-  skip_tests <- FALSE
   if (is.null(element_id)) {
     stop(
       "The variable ", level,
       "_id is NULL, please provide a valid ", level,
       "_id."
     )
-  } else {
-    if (level == "location") {
-      locations <- element_id |>
-        stringr::str_split("\\|")
-      if (any(locations |> sapply(length) < 3)) {
-        stop('Invalid location IDs found, these should be of the form "XXX|xxxx|1b3d5".')
-      } else {
-        # Extract the individual 5 digit location IDs
-        locations <- locations |>
-          as.data.frame() |>
-          t() |>
-          as.data.frame() |>
-          dplyr::rename(level = "V1", type = "V2", value = "V3")
-        element_id_proc <- locations$value
-        if (all(locations$type == "id")) {
-          example_vector <- eesyapi::example_id("location") |>
-            stringr::str_split("\\|")
-          example_id_string <- example_vector[[1]][3]
-        } else {
-          skip_tests <- TRUE
-        }
-      }
+  }
+  if (level == "location") {
+    locations <- element_id |>
+      stringr::str_split("\\|", simplify = TRUE)
+    if ("" %in% locations) {
+      stop('Invalid locations found, these should be of the form "LEVEL|xxxx|1b3d5".')
     } else {
-      element_id_proc <- element_id
-      example_id_string <- eesyapi::example_id(level)
-    }
-    if (!skip_tests) {
-      if (any(stringr::str_length(element_id) != stringr::str_length(eesyapi::example_id(level)))) {
-        err_string <- paste0(
-          "The ", level,
-          "_id(s) provided (", paste0(element_id_proc, collapse = ", "),
-          ") is expected to be a ",
-          stringr::str_length(example_id_string),
-          " character string in the format:\n    ",
-          example_id_string,
-          "\n  Please double check your ", level,
-          "_id."
-        )
-        stop(err_string)
-      } else if (
-        any(
-          gsub("[0-9a-zA-Z]", "", element_id_proc) != gsub("[0-9a-zA-Z]", "", example_id_string)
-        )
-      ) {
-        stop(err_string)
+      # Extract the individual 5 digit location IDs
+      df_locations <- locations |>
```

Comment:
Think you've got a bug somewhere in here? Doesn't handle if someone just puts one element in
```
> validate_ees_id("qwery", level = "location")
Error in `dplyr::rename()` at eesyapi/R/validation_rules.R:60:7:
! Can't rename columns that don't exist.
✖ Column `V2` doesn't exist.
Run `rlang::last_trace()` to see where the error occurred.
> rlang::last_trace()
<error/vctrs_error_subscript_oob>
Error in `dplyr::rename()` at eesyapi/R/validation_rules.R:60:7:
! Can't rename columns that don't exist.
✖ Column `V2` doesn't exist.
---
Backtrace:
    ▆
 1. └─eesyapi::validate_ees_id("qwery", level = "location")
 2.   ├─dplyr::rename(...) at eesyapi/R/validation_rules.R:60:7
 3.   └─dplyr:::rename.data.frame(...)
Run rlang::last_trace(drop = FALSE) to see 18 hidden frames.
```

Reply:
Should be fixed now